File: apps/ow-node-manage-api/app/model/database/column.js
/*
 * @Author: 钟凯
 * @Date: 2021-03-01 14:58:53
 * @LastEditTime: 2021-03-31 15:37:49
 * @LastEditors: 钟凯
 * @Description:
 * @FilePath: \egg_ts\app\model\database\column.ts
 * @可以输入预定的版权声明、个性签名、空行等
 */
'use strict';
const NameCodeModel = require('../base_name_code');


module.exports = app => {
  const { STRING, TEXT, BOOLEAN, INTEGER, BIGINT } = app.Sequelize;

  const Column = app.model.define('column', {
    ...NameCodeModel,
    cnUid: { type: STRING, allowNull: false, comment: '连接uid' },
    dbUid: { type: STRING, allowNull: false, comment: '数据库uid' },
    tbUid: { type: STRING, allowNull: false, comment: '数据库表uid' },
    dbName: { type: STRING, allowNull: false, comment: '数据库uid' },
    type: { type: TEXT, allowNull: false, comment: '类型' },
    isNull: { type: BOOLEAN, allowNull: false, comment: '是否为空' },
    isUnique: { type: BOOLEAN, allowNull: false, comment: '是否主键' },
    length: { type: BIGINT, allowNull: true, comment: '长度' },
    precision: { type: INTEGER, allowNull: true, comment: '精度' },
    oredr: { type: INTEGER, allowNull: true, comment: '序号' },
    defaultValue: { type: TEXT, allowNull: true, comment: '默认值' },
    description: { type: TEXT, allowNull: true, comment: '描述' },

  }, {
    tableName: 'ow_database_columns',
    paranoid: true,
  });
  return Column;
};

File: apps/onework-tools/onework-tools-webapi/src/main/java/com/onework/tools/database/Column.java
package com.onework.tools.database;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

/**
 * @projectName: onework-tools
 * @package: com.onework.tools.database
 * @className: DataBaseColumn
 * @author: 钟凯
 * @description: 数据库表字段
 * @date: 2021/12/7 23:35
 * @version: 1.0
 */
@Data
public class Column {

  /**
   * 数据库表
   */
  @NonNull
  private Table table;

  /**
   * 类型
   */
  @NonNull
  private String type;

  /**
   * 是否为空
   */
  @NonNull
  private Boolean isNull;

  /**
   * 是否主键
   */
  @NonNull
  private Boolean isUnique = false;

  /**
   * 长度
   */
  private Long length;

  /**
   * 精度
   */
  private Integer precision;

  /**
   * 序号
   */
  private Integer order;

  /**
   * 默认值
   */
  private String defaultValue;

  /**
   * 描述
   */
  private String description;
}
